### ISSUE TITLE
Disabling descendants

### ISSUE THREAD

**Issue by rkernahan** [clientside/amplesdk#19]
$isAccessible has been implemented in XUL and XHTML in this way:
        return !this.attributes["disabled"];

While this disables the immediate element, it does not disable descendants, which is highly desirable. If an element checks recursively for a disabled parent, we can ensure that all descendants are also disabled.

```
    StandardWidget.prototype.$isAccessible = function() {
        // fixme: to work, this same logic must be applied to XHTML elements!
        var oParent = this.parentNode;
        if (!oParent || oParent === ample.lastChild) {
            return(true);
        }
        if (!oParent.$isAccessible()) {
            return(false);
        }
        return(this.getAttribute('disabled') != 'true');    
    };
```

Similar code may be added to each language (XUL, XHTML).
If it is not added, then wrapping in a DIV would re-enable the DIV's descendants.

Could this ancestor-checking logic instead be added to AMLElement so that it is the default behavior for all languages?

**Comment by clientside** [comment]
See the problem. I will approach it differently though - from the core. (Doing the suggested way is expensive) 
Actually there is already functionality in the core that is supposed to do the trick http://github.com/clientside/amplesdk/blob/master/ample/runtime/classes/core/AMLNode.js#L669 somehow it doesn't work. Note, only UIEvent events will be skipped from handling on disabled elements and their children towards target. Any other events will still be handled.

**Comment by clientside** [comment]
Fixed in commit cab36b93d50c38e6338b

**Comment by clientside** [comment]
Another issue introduced in commit needs to be fixed

**Comment by clientside** [comment]
Fixed per commit ff8d23cab41df08db0cc
